src/AlgebraicGeometry/ToricVarieties/NormalToricVarieties/attributes.jl

```diff
@@ -861,10 +893,8 @@ Abelian group with structure: Z
         throw(ArgumentError("Group of the torus-invariant Cartier divisors can only be computed if the variety has no torus factor"))
     end
     
-    # compute mapping
-    map1 = map_from_torusinvariant_cartier_divisor_group_to_torusinvariant_weil_divisor_group(v)
-    map2 = map_from_torusinvariant_weil_divisor_group_to_class_group(v)
-    return restrict_codomain(map1*map2)
+    f = restrict_codomain(map_from_torusinvariant_cartier_divisor_group_to_class_group(v))
+    return f * inv(snf(codomain(f))[2])
```

Comment:
Why is `inv(snf(codomain(f))[2]` necessary? It seems the original code did not have this.

Reply:
You're right, the original code did not include `snf` here. But it did include it for the [class group](https://github.com/oscar-system/Oscar.jl/blob/4690df274f4113fc9e5898c839a133b3033b5c35/src/AlgebraicGeometry/ToricVarieties/NormalToricVarieties/attributes.jl#L731). The class group was always returned in smith normal form but the Picard group was not. I propose return both of them in snf, especially since I also add `map_from_picard_group_to_class_group`.

Reply:
I see. Good catch! That is appreciated.